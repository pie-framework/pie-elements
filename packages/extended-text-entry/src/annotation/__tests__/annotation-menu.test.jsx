import { shallow } from 'enzyme';
import React from 'react';
import AnnotationMenu from '../annotation-menu';

describe('freeform editor', () => {
  const defaultProps = {
    classes: {},
    anchorEl: {},
    open: false,
    annotations: [],
    isNewAnnotation: false
  };
  let onClose;
  let onDelete;
  let onEdit;
  let onWrite;
  let onAnnotate;

  beforeEach(() => {
    onClose = jest.fn();
    onDelete = jest.fn();
    onEdit = jest.fn();
    onWrite = jest.fn();
    onAnnotate = jest.fn();
  });

  const wrapper = extras => {
    const defaults = {
      ...defaultProps,
      onClose,
      onDelete,
      onEdit,
      onWrite,
      onAnnotate
    };
    const props = { ...defaults, ...extras };

    return shallow(<AnnotationMenu { ...props } />);
  };

  describe('snapshots', () => {
    it('renders', () => {
      expect(wrapper()).toMatchSnapshot();
    });

    it('opened', () => {
      expect(wrapper({ open: true })).toMatchSnapshot();
    });

    it('opened and new annotation selected', () => {
      expect(
        wrapper({ open: true, isNewAnnotation: true })
      ).toMatchSnapshot();
    });
  });
});
