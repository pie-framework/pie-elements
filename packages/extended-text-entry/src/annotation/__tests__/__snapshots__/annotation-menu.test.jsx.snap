// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`freeform editor snapshots opened 1`] = `
<AnnotationMenu
  anchorEl={Object {}}
  annotations={Array []}
  classes={
    Object {
      "annotationsWrapper": "AnnotationMenu-annotationsWrapper-2",
      "arrow": "AnnotationMenu-arrow-8",
      "button": "AnnotationMenu-button-4",
      "controlsWrapper": "AnnotationMenu-controlsWrapper-3",
      "holder": "AnnotationMenu-holder-7",
      "mainWrapper": "AnnotationMenu-mainWrapper-1",
      "negative": "AnnotationMenu-negative-6",
      "positive": "AnnotationMenu-positive-5",
    }
  }
  isNewAnnotation={false}
  onAnnotate={[MockFunction]}
  onClose={[MockFunction]}
  onDelete={[MockFunction]}
  onEdit={[MockFunction]}
  onWrite={[MockFunction]}
  open={true}
/>
`;

exports[`freeform editor snapshots opened and new annotation selected 1`] = `
<AnnotationMenu
  anchorEl={Object {}}
  annotations={Array []}
  classes={
    Object {
      "annotationsWrapper": "AnnotationMenu-annotationsWrapper-2",
      "arrow": "AnnotationMenu-arrow-8",
      "button": "AnnotationMenu-button-4",
      "controlsWrapper": "AnnotationMenu-controlsWrapper-3",
      "holder": "AnnotationMenu-holder-7",
      "mainWrapper": "AnnotationMenu-mainWrapper-1",
      "negative": "AnnotationMenu-negative-6",
      "positive": "AnnotationMenu-positive-5",
    }
  }
  isNewAnnotation={true}
  onAnnotate={[MockFunction]}
  onClose={[MockFunction]}
  onDelete={[MockFunction]}
  onEdit={[MockFunction]}
  onWrite={[MockFunction]}
  open={true}
/>
`;

exports[`freeform editor snapshots renders 1`] = `
<AnnotationMenu
  anchorEl={Object {}}
  annotations={Array []}
  classes={
    Object {
      "annotationsWrapper": "AnnotationMenu-annotationsWrapper-2",
      "arrow": "AnnotationMenu-arrow-8",
      "button": "AnnotationMenu-button-4",
      "controlsWrapper": "AnnotationMenu-controlsWrapper-3",
      "holder": "AnnotationMenu-holder-7",
      "mainWrapper": "AnnotationMenu-mainWrapper-1",
      "negative": "AnnotationMenu-negative-6",
      "positive": "AnnotationMenu-positive-5",
    }
  }
  isNewAnnotation={false}
  onAnnotate={[MockFunction]}
  onClose={[MockFunction]}
  onDelete={[MockFunction]}
  onEdit={[MockFunction]}
  onWrite={[MockFunction]}
  open={false}
/>
`;
